### PR TITLE
CONCF-672 | CuratedContent/GameGuides setting/purging cache

### PR DESCRIPTION
**Pull request by Warkot**
It should be compatible with https://github.com/Wikia/mercury/pull/1001, but it seems that mercury is not so keen on caching these days. Nevertheless, stuff has to be purged, if it's cached it's changing afterwards.

Please CR. @Wikia/x-wing @owend @macbre 

:cat2:

### CODE REVIEW

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ public function getPage() {
 		//This will always return json
 		$this->response->setFormat( 'json' );
 
-		$this->cacheResponseFor( 7, self::DAYS );
+		$this->cacheResponseFor( 24, self::HOURS );
```
Comment (macbre) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: Why can't we call `setCacheValidity` here with one of the constant values taken from `WikiaResponse`?

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 2]
```diff
@@ -400,12 +392,61 @@ function getJsonItem( $titleName, $ns, $pageId ) {
...
+						$wgServer .
+						MercuryApiHooks::SERVICE_API_BASE .
+						MercuryApiHooks::SERVICE_API_CURATED_CONTENT .
+						self::encodeURIQueryParam( $item[ 'title' ] );
+				}
+				return $urls;
+			},
+			// Purge all sections list getter URL - no additional params
+			[ self::getUrl( 'getList' ) ]
+		) ) ) )->doUpdate();
```
Comment (macbre) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: As we're purging four different URLs here maybe it's worth to consider to use surrogate keys here instead.

File: extensions/wikia/CuratedContent/CuratedContentController.class.php (hunk 1 above)
Comment (Warkot) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: I just left it how it was before, but indeed, it's a good idea. :+1:

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 3]
```diff
@@ -400,12 +392,62 @@ function getJsonItem( $titleName, $ns, $pageId ) {
...
 	static function onCuratedContentSave() {
-		self::purgeMethod( 'getList' );
+		global $wgServer;
+
+		$content = F::app()->wg->WikiaCuratedContent;
+
+		( new SquidUpdate( array_unique( array_reduce(
+			$content,
+			function ( $urls, $item ) use ( $wgServer ) {
+				if ( $item[ 'title' ] !== '' && empty( $item[ 'featured' ] ) ) {
```
Comment (bkoval) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: Why not to use empty() for the first check?

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 4]
```diff
@@ -400,12 +392,62 @@ function getJsonItem( $titleName, $ns, $pageId ) {
...
+					// Purge section URLs using urlencode() (standard for MediaWiki), which uses implements RFC 1738
+					// https://tools.ietf.org/html/rfc1738#section-2.2 - spaces encoded as `+`.
+					// iOS apps use this variant.
+					$urls[ ] = self::getUrl( 'getList' ) . '&section=' . urlencode( $item[ 'title' ] );
```
Comment (bkoval) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: Is the space inside empty square brackets forced by the guidelines? :)

File: extensions/wikia/CuratedContent/CuratedContentController.class.php (hunk 3 above)
Comment (Warkot) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: `empty` would return `true` when `$item[ 'title' ]` is `false`, `null`, empty array etc.

File: extensions/wikia/CuratedContent/CuratedContentController.class.php (hunk 4 above)
Comment (Warkot) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: Hard to tell, it's not a definition, but it's also not strictly indexing.  https://github.com/Wikia/guidelines/blob/master/PHP/CodingConventions.md#array-syntax
I would rather go without spaces inside, as this construct is closer to indexing. I wrote it this way to be consistent with the rest of the file.

**Comment by Warkot** [comment]
As a reminder — not only API calls have to be purged, but also URLs to the new Curated Main Pages routes like `/main/section/`, `/main/category/`.

**Comment by Warkot** [comment]
It's continued in #7793.

File: extensions/wikia/CuratedContent/CuratedContentController.class.php (hunk 2 above)
Comment (Warkot) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: Is it possible to purge internal cache using surrogate keys — for responses like from Mercury's Hapi.js to Mercury API in MediaWiki?
We could move our conversation to new PR: #7793.

**Pull request closed without merging**